MovingBlocks/Terasology
# Add feature to export/import multiplayer identities

### Contains

Implements the feature suggested in #1433; in other words, adds two buttons in the player settings screen that allow to export the identities used to authenticate to multiplayer server to a file, and to import a file of previously exported identities.
A NUI-based file picker is also included to allow the user to choose the file to save to or load from (screenshot attached).
![screenshot_20171015_191840](https://user-images.githubusercontent.com/15270303/31587204-ae869300-b1dd-11e7-9cd7-7305e9a1f391.png)

### How to test

* Make sure to have some identities in your configuration (if not sure, just join a server to automatically generate one). For testing purposes, also make sure that at least one identity is associated with something meaningful; for example, join a server and move your character to a certain place and/or pick up or drop some items.
* From the main menu, go to Settings > Player, and click the `Export...` button next to the `Multiplayer identities` label. the file picker popup will open; navigate to the directory where you want the file containing the identities to be saved, then enter a name for the file in the text box at the bottom of the popup and click OK.
* Exit Terasology and delete or rename the configuration file (`config.cfg`); obviously before deleting it, back it up if it contains something you want to preserve beside identities.
* (optional) Start the game again and join the same server; you will be placed at the spawn point and have the default item set because the identity has been deleted when you deleted the configuration file.
* Go back to the main menu, then go to Settings > Player and click `Import...`; in the file picker, select the file you previously exported to and click OK. If you did the previous step, you will be prompted to choose if you want to replace the locally saved identity; choose to import it anyway, overwriting the one which has been generated while performing the previous step with the one present in the file.
* Join the same server; you will be in the same position, have the same items, etc. as before deleting the configuration file.

### Outstanding before merging

Nothing I can think of, but obviously any feedback/suggestion is welcome.
